index.ts

```diff
@@ -5,26 +5,44 @@ import {
   ODataValueContextOfIListOfEntry,
   RepositoryInfo,
 } from '@laserfiche/lf-repository-api-client';
-import { OAuthAccessKey, servicePrincipalKey, repoId } from './ServiceConfig.js';
+import {
+  OAuthAccessKey,
+  servicePrincipalKey,
+  repositoryId,
+  authorizationType,
+  username,
+  password,
+  baseUrl,
+} from './ServiceConfig.js';
+import { authorizationTypeEnum as authType } from './AuthorizationType.js';
 import 'isomorphic-fetch';
 
 //Create a Laserfiche Repository API Client
-const _RepositoryApiClient: IRepositoryApiClient = createRepoAPIClient();
+let _RepositoryApiClient: IRepositoryApiClient;
 const rootFolderEntryId = 1;
 
 await main();
 
 async function main(): Promise<void> {
   try {
-    await getRepoName(); //Print repository name
+    if (authorizationType === authType.CloudAccessKey) {
+      _RepositoryApiClient = createCloudRepoApiClient();
+    } else if (authorizationType === authType.APIServerUsernamePassword) {
+      _RepositoryApiClient = createSelfHostedRepoApiClient();
+    } else {
+      console.error(
+        "Invalid value for 'AUTHORIZATION_TYPE'. It can only be 'CloudAccessKey' or 'APIServerUsernamePassword'."
```

Comment:
the valid enum values seems to be CLOUD_ACCESS_KEY and API_SERVER_USERNAME_PASSWORD

Reply:
done